doc/changes/latest.rst

```diff
@@ -72,3 +72,6 @@ Changes
 - Convert references in ``nilearn/glm/regression.py`` and ``nilearn/glm/thresholding.py`` to use footcite/footbibliography (:gh:`3302` by `Ahmad Chamma`_).
 - Boolean input data in :func:`~image.new_img_like` now defaults to `np.uint8` instead of `np.int8` (:gh:`3286` by `Yasmin Mzayek`_).
 - The current behavior of maskers' ``transform`` on 3D niimg inputs, in which a 2D array is returned, is deprecated, and 1D arrays will be returned starting in version ``0.12`` (:gh:`3322` by `Taylor Salo`_).
+- :func:`~regions.rena_clustering.weighted_connectivity_graph` is changed to a private function :func:`~regions.rena_clustering._weighted_connectivity_graph`
```

Comment:
These references won't work since the functions are private or don't exist since you renamed them. Also, it's much easier to read the changelog when there is one entry per PR (if it is possible).

I would go for something like this, deleting the next two lines (WDYT?): 

```suggestion
- Private functions ``nilearn.regions.rena_clustering.weighted_connectivity_graph`` and ``nilearn.regions.rena_clustering.nearest_neighbor_grouping`` have been renamed with a leading "_", while function :func:`~regions.rena_clustering.recursive_neighbor_agglomeration` has been added to the public API (:gh:`3347` by `Ahmad Chamma`_).
```